packages/samples/headless-commerce-react/src/components/facets/facet-generator/facet-generator.tsx

```diff
@@ -0,0 +1,46 @@
+import {FacetGenerator as HeadlessFacetGenerator} from '@coveo/headless/commerce';
+import {useEffect, useState} from 'react';
+import CategoryFacet from '../category-facet/category-facet';
+import DateFacet from '../date-facet/date-facet';
+import NumericFacet from '../numeric-facet/numeric-facet';
+import RegularFacet from '../regular-facet/regular-facet';
+
+interface IFacetGeneratorProps {
+  controller: HeadlessFacetGenerator;
+}
+
+export default function FacetGenerator(props: IFacetGeneratorProps) {
+  const {controller} = props;
+
+  const [state, setState] = useState(controller.state);
+  const facets = controller.facets;
```

Comment:
Don't we need to use a `useState` hook to contain the facets here as well? As state changes occur, in the the `useEffect` below, we should update the facets.